docs/conceptual/iommu.rst

```diff
@@ -0,0 +1,98 @@
+at two devices and determine whether or not they can support peer to peer DMA. This functionality is controlled
+by the ``CONFIG_PCI_P2PDMA`` kernel config option. In addition to that parameter, you also need to enable
+``dma-buf`` (the framework for sharing DMA memory across drivers). To properly support peer to peer DMA with the IOMMU,
+the following options must be enabled in your kernel config: ``CONFIG_PCI_P2PDMA``, ``CONFIG_DMABUF_MOVE_NOTIFY`` 
+and ``CONFIG_HSA_AMD_P2P``.
+
+Peer to Peer DMA over XGMI
+----------------------------------------------------------------
+
+Devices connected via XGMI links (MI50/60, MI100, MI200, MI300) have private XGMI links between GPUs so the
+PCIe bus is not used for peer to peer DMA. As such, device physical addressing limitations and platform root
+port limitations do not come into play. Note that non-GPU devices, such as RDMA NICs still use PCIe for peer
+to peer DMA so the limitations mentioned in the section above would apply to interactions between the GPUs and those devices.
+
+Linux IOMMU support
+================================================================
+
+Linux has supported the IOMMU for more than a decade in both virtualized environments and bare metal.
+There are 3 modes of operation most commonly used:
+
+* IOMMU disabled. The IOMMU is disabled and the entire platform shares a common platform address space for 
+  system memory and MMIO spaces. Setting ``iommu=off`` on the kernel command line will disable the IOMMU.
+
+* IOMMU enabled in passthrough mode. In this case, the IOMMU is enabled, but the identity mapping is used for all devices.
+  The IOMMU is enabled, but the entire platform still shares a common platform address space for system memory and MMIO spaces.
+  Setting ``iommu=pt`` on the kernel command line enables this mode.  There are cases where you many need some
+  aspects of the IOMMU (e.g., interrupt remapping), but don't want DMA remapping or you have a system with a lot of cores and
+  want good interrupt distribution across the cores.
+
+* IOMMU enabled in remapping mode. In this mode, the IOMMU is enabled and each device gets it's own IO virtual address space.
+  All devices on Linux register their DMA addressing capabilities (e.g. 32 bit, 40 bit, 44, bit, etc. i.e., how much address
+  space can they natively access). The kernel will then make sure that any address space mapped for DMA is mapped within
+  the DMA addressing limits of the device. Additionally, only address space explicitly mapped by the device is mapped into
+  its virtual address space so an access to an unmapped page will generate an IOMMU page fault. This is generally the
+  default when the IOMMU is enabled.
+
+Advantages and disadvantages
+================================================================
+
+There advantages and disadvantages depending on the devices and platforms you are using.
```

Comment:
Advantages and disadvantages to what? To using an IOMMU?

Reply:
Is it that the IOMMU can be enabled/disabled?

Reply:
Yup, IOMMU has 3 modes, disabled, passthrough mode and remapping mode, where the latter two are considered enabled. I believe the advantage and disadvantage are regarding to enable/disable IOMMU.

Reply:
I think the point is to provide some context around IOMMU in general.  A lot of users don't know what it is and there is a lot of conflicting information about it.  For various historical reasons, AMD had recommended that users disable it for either compatibility or performance reasons for ROCm.  Then as the number of cores on our CPUs increased we had to temper our statement to make sure systems with a lot of cores worked properly.  So we recommend iommu=pt rather than iommu=off.  As such, the level of validation we do with the IOMMU fully enabled has always been more limited.  In the long term however, it's an important compatibility and security tool that we should embrace and encourage.  It protects devices from buggy DMAs from other devices and it allows for P2P DMA on systems with large numbers of devices which may not be possible with the system address space layouts on most modern systems.  So now we are at another crossroads where we need to reconsider our recommended stance on IOMMU settings.